### PR TITLE
Make Gardener audit configuration configurable in Helm chart

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**: Make Gardener audit configuration configurable in Helm chart

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: charts/gardener/charts/runtime/templates/deployment-apiserver.yaml [hunk 1]
```diff
@@ -19,7 +19,12 @@ spec:
   template:
     metadata:
       annotations:
-        checksum/configmap-gardener-audit-policy-config: {{ include (print $.Template.BasePath "/configmap-apiserver-audit.yaml") . | sha256sum }}
+        {{- if .Values.global.apiserver.audit.policy }}
+        checksum/configmap-gardener-audit-policy-config: {{ include (print $.Template.BasePath "/configmap-apiserver-audit-policy.yaml") . | sha256sum }}
+        {{- end }}
+        {{- if .Values.global.apiserver.audit.policy }}
```
Comment (vpnachev) [inline charts/gardener/charts/runtime/templates/deployment-apiserver.yaml]:
```suggestion
        {{- if .Values.global.apiserver.audit.webhook.config }}
```

The `webhook config` does not depend on the policy file.

**Review by vpnachev — CHANGES_REQUESTED** [review]
Thanks!
I propose a small change in the deployment manifest of the `gardener-apiserver`.

File: charts/gardener/charts/runtime/templates/deployment-apiserver.yaml (hunk 1 above)
Comment (rfranzke) [reply charts/gardener/charts/runtime/templates/deployment-apiserver.yaml]: nice catch

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**